webapp/packages/plugin-sql-editor/src/SqlEditor/useSqlEditor.ts

```diff
@@ -228,7 +226,7 @@ export function useSqlEditor(state: ISqlEditorTabState): ISQLEditorData {
           const formatted = await this.sqlDialectInfoService.formatScript(this.dataSource.executionContext, script.query);
 
           this.onFormat.execute([script, formatted]);
-          this.setQuery(query.substring(0, script.begin) + formatted + query.substring(script.end));
+          this.setScript(query.substring(0, script.begin) + formatted + query.substring(script.end));
```

Comment:
Sometimes JS behaves crazy in this cases of concatenation if data is not as it expected to be. I would recommend to use `.concat` method to avoid bugs in future